src/main/kotlin/lotto/dto/LottoResult.kt

```diff
@@ -0,0 +1,21 @@
+package lotto.dto
+
+import lotto.vo.LottoNumber
+
+data class LottoResult private constructor (private val _winningNumber: List<LottoNumber>) {
+    val winningNumber
+        get() = _winningNumber
+
+    companion object {
+        private const val WINNING_NUMBERS_DELIMITER = ","
+
+        fun of(winningNumbers: String?): LottoResult {
+            require(winningNumbers != null) { "당첨 번호는 null이 될 수 없습니다." }
+
+            winningNumbers.split(WINNING_NUMBERS_DELIMITER)
+                .map { n -> n.toInt() }
+                .map { n -> LottoNumber(n) }
+                .let { n -> return LottoResult(n) }
```

Comment:
개인적으로 아래와 같이 작성하는 것이 조금 더 가독성이 좋다고 생각됩니다. 😃 (n이라는 의미가 애매한 람다 변수를 쓰는 것보다는 차라리 `it`을 활용하는 것이 좋다고 생각해요 😉)

또한 map 함수에서 사용한 람다 변수 `n`과  **let scope function**에서 사용한 `n`은 완전히 다른 타입인데, 같은 변수명으로 쓰다보니 이 차이를 바로 이해하는데 시간이 걸렸어요. 👀 또한 scope function의 람다식 안에서 바로 return 하는 부분도 한 눈에 캐치하기가 힘들었던 것 같습니다. 🤔

```suggestion
            return winningNumbers.split(WINNING_NUMBERS_DELIMITER)
                .map { it.toInt() }
                .map { LottoNumber(it) }
                .let { LottoResult(it) }
```

Reply:
- 명확한 return 문 
- 명확한 의미의 람다 변수 / `it` 사용 

적용 완료했습니다!